kiwi_post_run

```diff
@@ -1,188 +1,264 @@
+		       provides excludearches targetboot kernelboot \
+		       initrdboot ignoreimage targetimage image \
+		       md5image kernelimage initrdimage bootoptions \
+		       append; do
+	_cfg[$key]=$(get_xml_data "string(//props/$key)" ${properties_xml})
+    done
+
+    declare -A default=(
+	['licensefile']='LICENSE'
+	['targetboot']="/srv/${_cfg[name]}"
+	['kernelboot']='linux'
+	["initrdboot"]="initrd"
+	["ignoreimage"]="false"
+	["image"]="${_cfg[name]}-${_cfg[arch]}-${_cfg[version]}.xz"
+	["md5image"]="${_cfg[name]}-${_cfg[arch]}-${_cfg[version]}.md5"
+	['kernelimage']="${_cfg[name]}-${_cfg[arch]}-${_cfg[version]}.kernel"
+	["initrdimage"]="${_cfg[name]}-${_cfg[arch]}-${_cfg[version]}.initrd"
+	["bootoptions"]="${_cfg[name]}-${_cfg[arch]}-${_cfg[version]}.config.bootoptions"
+	["append"]="${_cfg[name]}-${_cfg[arch]}-${_cfg[version]}.append"
+    )
+    for key in "${!_cfg[@]}"; do
+	if [ "${_cfg[$key]}" == "''" -o -z "${_cfg[$key]}" ]; then
+	    _cfg[$key]="${default[$key]}"
+	fi
+    done
+
+    # Some defaults depend on other default values, so we need a last
+    # iteration to resolve them
+    declare -A default=(
+	["targetimage"]="${_cfg[targetboot]}/image"
+    )
+    for key in "${!_cfg[@]}"; do
+	if [ "${_cfg[$key]}" == "''" -o -z "${_cfg[$key]}" ]; then
+	    _cfg[$key]="${default[$key]}"
+	fi
+    done
+}
 
-# Some other required variables
-RELEASE=$(date +%Y%m%d%H%M%S)
-YEAR=$(date +%Y)
+expand_cfg() {
+    local -n _cfg="${1}"
+    if [ "${_cfg[url]}" != "''" -o ! -z "${_cfg[url]}" ]; then
+	_cfg['url']="URL:            ${_cfg[url]}"
+    fi
 
-# Read properties from rpm-properties.xml
-PROPERTIES_XML="${RPM_SOURCE_DIR}/rpm-properties.xml"
-if [ ! -f ${PROPERTIES_XML} ]; then
-  echo "ERROR: No ${PROPERTIES_XML} file. Make sure you have one in your package"
+    local provides=""
+    for provide in "${_cfg[provides]}"; do
+	[ -z "${provide}" ] && continue
+	provides="${provides}Provides:       ${provide}\n"
+    done
+    _cfg['provides']="${provides}"
+
+    local excludearches=""
+    for arch in "${_cfg[excludearches]}"; do
+	[ -z "${arch}" ] && continue
+	excludearches="${excludearches}ExcludeArch:    ${arch}\n"
+    done
+    _cfg['excludearches']="${excludearches}"
+
+    if [ "${cfg[ignoreimage]}" == "true" ]; then
+	cfg['ignoreimage']=1
+    else
+	cfg['ignoreimage']=0
+    fi
+}
+
+
+echo '██████████████████████████████████████████████████████████████'
+echo '█                                                            █'
+echo '█  containment-rpm-pxe: Turn your KIWI PXE images into RPMs  █'
+echo '█       https://github.com/openSUSE/container-rpm-pxe        █'
+echo '█                                                            █'
+echo '██████████████████████████████████████████████████████████████'
+
+cd "${kiwi_dir}"
+
+echo 'INFO: Looking for a tar file from KIWI ...'
+source=$(find_source)
+if [ -z "${source}" ]; then
+  echo 'ERROR: No tar file from KIWI found!'
   exit 1
```

Comment:
you could also do something like
```
  for i in *.install.tar *.tar.xz; do
  if [ -n "$source" ]; then
    error...
  else
    source="$f"
  fi
  done
```

Reply:
needs the nullglob shopt though or another test -e

Reply:
Not sure that is doing the same? For OEM we have the install.tar, and for PXE the tar.xz. If I understand the code correctly it will fail to detect if both are missing, erroring only if both are present.

Reply:
yeah, need another check afterwards